src/Data/BTree/Primitives/Height.hs

```diff
@@ -1,14 +1,17 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE KindSignatures #-}
 
 module Data.BTree.Primitives.Height where
 
+import Data.Binary (Binary)
+
 --------------------------------------------------------------------------------
 
 data Nat = Z | S Nat
 
 newtype Height (h :: Nat) = Height { fromHeight :: Int }
```

Comment:
You're serializing an Int here. What's the serialized size on IA32 and AMD64? Does it differ? I just checked the binary source and it's serialized as big-endian Int64. Maybe you can add a small test case that makes sure this assumption is never broken. In particular when switching to a different serialization library!!